packages/flutter_tools/lib/src/runner/flutter_command.dart

```diff
@@ -54,13 +51,16 @@ abstract class FlutterCommand extends Command {
     return await runInProject();
   }
 
-  bool validateProjectRoot() {
+  Validator validateProjectRoot = () {
```

Comment:
This was a bit confusing to me, but It looks you've changed it to a field so that you can modify the value for testing. Perhaps rename the field to something like `commandValidator`?